cmd/jujud-controller/agent/machine/manifolds.go

```diff
@@ -915,6 +886,30 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 			Clock:                    config.Clock,
 			Logger:                   internallogger.GetLogger("juju.worker.storageregistry"),
 		}),
+
+		httpClientName: httpclient.Manifold(httpclient.ManifoldConfig{
+			NewHTTPClient: func(namespace corehttp.Namespace, opts ...internalhttp.Option) *internalhttp.Client {
```

Comment:
> Shouldn't we be passing in opts to the client constructors below - I guess this will be in a follow up.

You are 100% correct.

I was waiting for providers to be implemented.